platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ImportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration.admin;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.configuration.persistence.PersistenceStrategy;
+import org.codice.ddf.migration.ImportMigrationContext;
+import org.codice.ddf.migration.ImportMigrationContextProxy;
+import org.codice.ddf.migration.ImportMigrationEntry;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationReport;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class extends on the {@link ImportMigrationContext} interface to pre-process exported entries
+ * for configuration objects and compare them with the configuration objects currently in memory.
+ */
+public class ImportMigrationConfigurationAdminContext extends ImportMigrationContextProxy {
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            ImportMigrationConfigurationAdminContext.class);
+
+    private final ConfigurationAdminMigratable admin;
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    /**
+     * Keeps track of all managed services in memory that were not found in the export file so
+     * we know what to delete at the end.
+     */
+    private final Map<String, Configuration> memoryServices;
+
+    /**
+     * Keeps track of all managed service factories in memory that were not found in the export file
+     * so we know what to delete at the end.
+     */
+    private final Map<String, List<Configuration>> memoryFactoryServices;
+
+    private final Map<String, ImportMigrationConfigurationAdminEntry> exportedServices;
```

Comment:
We could have but as with the export context, since the design is to get all of them anyway, I preferred to pre-create them in the constructor and fail there if need be.